### PR TITLE
Feat/258 move key value table into monorepo

### PR DESCRIPTION
**Pull request by MadalinaRaicu**
- KeyValueTable is moved to ui-toolkit
- updated styling to use tailwind
- updated tailwind config
- used within token app
- added story

Screenshot 2022-04-29 at 11 32 58
Screenshot 2022-04-27 at 19 15 20

### CODE REVIEW

File: apps/token/src/routes/governance/components/proposal-change-table/proposal-change-table.tsx [hunk 1]
```diff
@@ -19,26 +16,26 @@ export const ProposalChangeTable = ({ proposal }: ProposalChangeTableProps) => {
   const terms = proposal.terms;
 
   return (
-    <KeyValueTable muted={true} data-testid="proposal-change-table">
-      <KeyValueTableRow>
+    <KeyValueTable data-testid="proposal-change-table">
+      <KeyValueTableRow muted={true}>
```
Comment (MadalinaRaicu) [inline apps/token/src/routes/governance/components/proposal-change-table/proposal-change-table.tsx:20]: muted and numerical tailwind styling has been added directly on `tr`,  `td` and `th` tags. This is why these props have been moved to KeyValueTableRows. I am looking into passing these props to the children.

File: apps/token/src/routes/claim/claim-flow.tsx [hunk 2]
```diff
@@ -159,37 +156,34 @@ export const ClaimFlow = ({
           <div>
             <KeyValueTable>
               <KeyValueTableRow>
-                <th>{t('Connected Ethereum address')}</th>
-                <td>{truncateMiddle(address)}</td>
+                {t('Connected Ethereum address')}
```
Comment (MadalinaRaicu) [inline apps/token/src/routes/claim/claim-flow.tsx:159]: `<th>` and `<td>` tags have been moved inside KeyValueTableRow and tailwind styling has been applied to them.

File: apps/token/src/routes/home/token-details/token-details-circulating.tsx [hunk 3]
```diff
@@ -37,8 +37,8 @@ export const TokenDetailsCirculating = ({
 }) => {
   const totalCirculating = sumCirculatingTokens(tranches);
   return (
-    <td data-testid="circulating-supply">
+    <span data-testid="circulating-supply">
```
Comment (MadalinaRaicu) [inline apps/token/src/routes/home/token-details/token-details-circulating.tsx:40]: spotted this. reverting it now.

**Review by dexturr — CHANGES_REQUESTED** [review]
Looking good, just a few minor things 👍

File: libs/ui-toolkit/src/index.ts [hunk 4]
```diff
@@ -21,6 +21,7 @@ export { VegaLogo } from './components/vega-logo';
 export { Tooltip } from './components/tooltip';
 export { Indicator } from './components/indicator';
 export { Card } from './components/card';
+export { KeyValueTable, KeyValueTableRow } from './components/key-value-table';
```
[the remark this answers is not in the document]
Comment (MadalinaRaicu) [reply libs/ui-toolkit/src/index.ts]: resolved in my last commit

File: libs/ui-toolkit/src/components/key-value-table/key-value-table.tsx [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+  children,
+  className,
+  muted,
+  numerical,
+  ...rest
+}: KeyValueTableRowProps) => {
+  return (
+    <tr
+      {...rest}
+      className={`flex flex-col sm:table-row border-b first:border-t border-white ${
```
[the remark this answers is not in the document]
Comment (MadalinaRaicu) [reply libs/ui-toolkit/src/components/key-value-table/key-value-table.tsx]: resolved in last commit

File: apps/token/src/routes/governance/components/proposal-votes-table/proposal-votes-table.tsx [hunk 6]
```diff
@@ -34,65 +31,65 @@ export const ProposalVotesTable = ({ proposal }: ProposalVotesTableProps) => {
...
-        <th>{t('numberOfAgainstVotes')}</th>
-        <td>{formatNumber(noVotes, 0)}</td>
+        {t('numberOfAgainstVotes')}
+        <span>{formatNumber(noVotes, 0)}</span>
       </KeyValueTableRow>
       <KeyValueTableRow>
-        <th>{t('yesPercentage')}</th>
-        <td>{formatNumber(yesPercentage, 2)}%</td>
+        {t('yesPercentage')}
+        <span>{formatNumber(yesPercentage, 2)}%</span>
```
[the remark this answers is not in the document]
Comment (MadalinaRaicu) [reply apps/token/src/routes/governance/components/proposal-votes-table/proposal-votes-table.tsx]: created a util function in last commit

File: libs/tailwindcss-config/src/theme.js [hunk 7]
```diff
@@ -22,6 +22,10 @@ module.exports = {
       95: 'rgba(255, 255, 255, 0.95)',
       100: 'rgba(255, 255, 255, 1.00)',
     },
+    grey: {
```
[the remark this answers is not in the document]
Comment (MadalinaRaicu) [reply libs/tailwindcss-config/src/theme.js:25]: black-40 or black-60 have low opacity and on dark theme they get invisible 
image

File: libs/ui-toolkit/src/components/key-value-table/key-value-table.tsx [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+  numerical,
+  ...rest
+}: KeyValueTableProps) => {
+  return (
+    <React.Fragment>
+      {title && <h3 className="mt-2.5 mb-1.25">{title}</h3>}
+      <table
+        data-testid="key-value-table"
+        {...rest}
+        className={`w-full border-collapse mb-2.5 [border-spacing:0] break-all ${
```
Comment (sam-keen) [inline libs/ui-toolkit/src/components/key-value-table/key-value-table.tsx:26]: Chore (blocking): Our custom Tailwind theme means the values you've used here with decimals won't work. The currently allowed spacings are defined in libs/tailwindcss-config/src/theme.js. If you need a custom value, use the square bracket syntax, i.e. mb-[0.3rem]. 

Question: not seen the syntax for what you've done here with `[border-spacing:0]`. What does this do?

**Review by sam-keen — CHANGES_REQUESTED** [review]

File: apps/token/src/routes/governance/components/proposals-list/proposals-list.tsx [hunk 9]
```diff
@@ -33,36 +30,34 @@ export const ProposalsList = ({ proposals }: ProposalsListProps) => {
...
-              {isFuture(new Date(proposal.terms.enactmentDatetime))
-                ? t('proposedEnactment')
-                : t('enactedOn')}
-            </th>
-            <td data-testid="governance-proposal-enactmentDate">
+            {isFuture(new Date(proposal.terms.enactmentDatetime))
+              ? t('proposedEnactment')
+              : t('enactedOn')}
+
+            <span data-testid="governance-proposal-enactmentDatetime">
```
[the remark this answers is not in the document]
Comment (MadalinaRaicu) [reply apps/token/src/routes/governance/components/proposals-list/proposals-list.tsx]: thanks for spotting this. updated in last commit.

File: apps/token/src/routes/redemption/home/vesting-table.tsx [hunk 10]
```diff
@@ -42,29 +39,29 @@ export const VestingTable = ({
...
         >
-          <th>{t('Vesting VEGA')}</th>
-          <td>{formatNumber(total)}</td>
+          <span>{t('Vesting VEGA')}</span>
+          {formatNumber(total)}
         </KeyValueTableRow>
         <KeyValueTableRow data-testid="vesting-table-locked">
-          <th>
+          <span>
             <div className="vesting-table__indicator-square vesting-table__indicator-square--locked"></div>
```
[the remark this answers is not in the document]
Comment (MadalinaRaicu) [reply apps/token/src/routes/redemption/home/vesting-table.tsx]: I have replaced the nested divs with spans in the last commit.

File: libs/ui-toolkit/src/components/key-value-table/key-value-table.tsx (hunk 8 above)
Comment (sam-keen) [reply libs/ui-toolkit/src/components/key-value-table/key-value-table.tsx]: Ah that's good to know, thanks!

File: libs/ui-toolkit/src/components/key-value-table/key-value-table.tsx [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+
+export const KeyValueTableRow = ({
+  children,
+  className,
+  muted,
+  numerical,
+  ...rest
+}: KeyValueTableRowProps) => {
+  const trClassName = classNames(
+    'flex flex-col sm:table-row border-b first:border-t border-white',
```
[the remark this answers is not in the document]
Comment (bglownia) [reply libs/ui-toolkit/src/components/key-value-table/key-value-table.tsx:66]: Text is also not visible ;(

**Review by bglownia — CHANGES_REQUESTED** [review]

File: README.md [hunk 12]
```diff
@@ -40,7 +41,7 @@ Run `nx serve my-app` for a dev server. Navigate to http://localhost:4200/. The
 
 ### Running tests
 
-Run `nx test my-app` to execute the unit tests with [Jest](https://jestjs.io), or `nx affected:test` to execute just unit tests affected by a change.
+Run `yarn nx run <my-app>-e2e:e2e --watch` to execute the unit tests with [Jest](https://jestjs.io) in watch mode. Use `nx affected:test` to execute just unit tests affected by a change.
```
Comment (dexturr) [inline README.md:44]:
```suggestion
Run `yarn nx run <my-app>-e2e:e2e` to execute the e2e tests with [cypress](https://docs.cypress.io/).  You can use the `--watch` flag to open the cypress tests UI in watch mode, see [cypress executor](https://nx.dev/packages/cypress/executors/cypress) for all CLI flags.

Run `nx test my-app` to execute the unit tests with [Jest](https://jestjs.io), or `nx affected:test` to execute just unit tests affected by a change. You can also use `--watch` with these test to run jest in watch mode, all CLI flags can be found in the [Jest executor documentation](https://nx.dev/packages/jest/executors/jest)
```

File: libs/ui-toolkit/src/components/key-value-table/key-value-table.tsx [hunk 13]
```diff
@@ -0,0 +1,83 @@
...
+    {
+      'border-grey/1 first:[border-top:none] last:[border-bottom:none]': muted,
+    },
+    className
+  );
+  const thClassName = `break-word text-left font-medium text-white uppercase align-top p-4`;
+  const tdClassName = classNames('align-top p-4 text-right text-grey', {
```
[the remark this answers is not in the document]
Comment (MadalinaRaicu) [reply libs/ui-toolkit/src/components/key-value-table/key-value-table.tsx:73]: thanks for the clarification, I have updated the colors in my last commit.

File: libs/ui-toolkit/src/components/key-value-table/key-value-table.tsx (hunk 11 above)
[the remark this answers is not in the document]
Comment (MadalinaRaicu) [reply libs/ui-toolkit/src/components/key-value-table/key-value-table.tsx]: thanks for the clarification. colors are updated in my last commit.

File: libs/tailwindcss-config/src/theme.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (MadalinaRaicu) [reply libs/tailwindcss-config/src/theme.js:25]: thanks for the clarification. colors are updated in the last commit.

File: libs/ui-toolkit/src/components/key-value-table/key-value-table.tsx [hunk 14]
```diff
@@ -0,0 +1,83 @@
...
+    'font-mono': numerical,
+  });
+
+  return (
+    <tr {...rest} className={trClassName}>
```
[the remark this answers is not in the document]
Comment (MadalinaRaicu) [reply libs/ui-toolkit/src/components/key-value-table/key-value-table.tsx]: on it 👍

File: libs/ui-toolkit/src/components/key-value-table/key-value-table.tsx [hunk 15]
```diff
@@ -0,0 +1,83 @@
...
+  title,
+  children,
+  className,
+  muted,
+  numerical,
+  ...rest
+}: KeyValueTableProps) => {
+  return (
+    <React.Fragment>
+      {title && <h3 className="mt-8 mb-4">{title}</h3>}
```
[the remark this answers is not in the document]
Comment (MadalinaRaicu) [reply libs/ui-toolkit/src/components/key-value-table/key-value-table.tsx]: in my last commit I am also using the headingLevel prop to update the `text-h*` class

File: libs/ui-toolkit/src/components/key-value-table/key-value-table.tsx [hunk 16]
```diff
@@ -0,0 +1,95 @@
...
+  headingLevel,
+  ...rest
+}: KeyValueTableProps) => {
+  const TitleTag: keyof JSX.IntrinsicElements = headingLevel
+    ? `h${headingLevel}`
+    : 'div';
+  return (
+    <React.Fragment>
+      {title && (
+        <TitleTag className={`text-h${headingLevel} mt-8 mb-4`}>
```
Comment (MadalinaRaicu) [inline libs/ui-toolkit/src/components/key-value-table/key-value-table.tsx:28]: `text-h${headingLevel}` is required to reflect the proper header styling.

File: libs/ui-toolkit/src/components/key-value-table/key-value-table.tsx (hunk 14 above)
[the remark this answers is not in the document]
Comment (MadalinaRaicu) [reply libs/ui-toolkit/src/components/key-value-table/key-value-table.tsx]: updated to use `dl`. but, I can't apply the `{...rest}` on `dl`. I hope it is ok to remove it for now.

**Review by bglownia — APPROVED** [review]

File: .prettierignore [hunk 17]
```diff
@@ -3,3 +3,4 @@
 /dist
 /coverage
 __generated__
+*.md
```
Comment (MadalinaRaicu) [inline .prettierignore:6]: format didn't like `md` files

**Review by dexturr — APPROVED** [review]

**Review by sam-keen — APPROVED** [review]

**Pull request merged**